cmd/stellar-rpc/internal/integrationtest/infrastructure/test.go

```diff
@@ -70,15 +72,22 @@ type TestConfig struct {
 	SQLitePath string
 	OnlyRPC    *TestOnlyRPCConfig
 	// Do not mark the test as running in parallel
-	NoParallel bool
+	NoParallel                bool
+	EnableCoreHTTPQueryServer bool
 }
 
 type TestCorePorts struct {
 	CorePort        uint16
 	CoreHTTPPort    uint16
 	CoreArchivePort uint16
-	// This only needs to be an unconflicting port
-	captiveCorePort uint16
+
+	// These only need to be unconflicting ports
+	captiveCorePort          uint16
+	captiveCoreHTTPQueryPort uint16
+	// TODO: this is only needed due to a quirk (bug) in the implementation of the
+	//       core's query server, which requires the http port to be set to non-zero for the query server to be spawned.
```

Comment:
Just to clarify, this means the "regular" HTTP server **must** be enabled for the "query" server to be enable-able, right?

Reply:
Yes. Feel free to rephrase it!

Reply:
No need, it's clear but I just wanted to make sure I understood! :+1: